Sources/Foundation/NSStringAPI.swift

```diff
@@ -74,9 +74,12 @@ internal func _persistCString(_ p: UnsafePointer<CChar>?) -> [CChar]? {
         return nil
     }
     let len = UTF8._nullCodeUnitOffset(in: cString)
-    var result = [CChar](repeating: 0, count: len + 1)
-    for i in 0..<len {
-        result[i] = cString[i]
+    let result = [CChar](unsafeUninitializedCapacity: len + 1) { buf, initedCount in
+        for i in 0..<len {
+            buf[i] = cString[i]
+        }
+        buf[len] = 0
+        initedCount = len + 1
```

Comment:
Would the copy be better as:
```
let bytesToCopy = UTF8._nullCodeUnitOffset(in: cString) + 1 // +1 for the terminating NUL
let result = [CChar](unsafeUninitializedCapacity: bytesToCopy) { buf, initedCount in
        buf.baseAddress!.assign(from: cString, count: bytesToCopy)
         initedCount = bytesToCopy
}
```

Reply:
You are right. Thank you for suggestion.